src/clp_ffi_js/ir/LogEventWithFilterData.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+#define CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+
+#include <utility>
+
+#include <clp/ir/types.hpp>
+
+#include <clp_ffi_js/constants.hpp>
+
+namespace clp_ffi_js::ir {
+/**
+ * A class which accepts a log event type as a template parameter and provides additional members
+ * for the log level and timestamp. The additional members facilitate log level filtering.
+ * @tparam LogEvent The type of the log event.
+ */
+template <typename LogEvent>
+class LogEventWithFilterData {
+public:
+    // Constructor
+    explicit LogEventWithFilterData(
```

Comment:
Since we have more than one parameters in this constructor, `explicit` is not meaningful here.
```suggestion
    LogEventWithFilterData(
```